static/app/utils/useTeams.tsx

```diff
@@ -172,8 +228,28 @@ function useTeams({limit}: Options = {}) {
     }
   }
 
+  useEffect(() => {
+    if (slugs) {
+      const teamsInStore = store.teams.filter(t => slugs.includes(t.slug));
+
+      // Only fetch the slugs if they do not exist in the store already
+      if (teamsInStore.length !== slugs.length) {
+        loadTeamsBySlug();
+      }
+    } else if (provideUserTeams && !store.loadedUserTeams) {
+      loadUserTeams();
+    }
+  }, [JSON.stringify(slugs), provideUserTeams]);
```

Comment:
not sure how weird it is to do something like this, but I don't want to continually refetch projects everytime `slugs` changes due to object comparison here.

Reply:
So calling the `JSON.stringify(slugs)` will cause this to change on every render I think, since it's not comparing value but comparing reference value right? Or does it actually compare string value?

Reply:
It should compare string values

Reply:
I can double check that tho

Reply:
To my best knowledge this works, but it definitely feels very hacky. A workaround might be to compare prevProps and currentProps by storing prevProps in a `useRef`.

Reply:
I think I would prefer to do that over this. The other issue with this is that order might break things.